openapiv2: panic when SecurityRequirement is missing a value

## 🐛 Bug Report

`protoc-gen-openapiv2` panics when specifying an OpenAPI security option with a security requirement that is missing a value.

This was discovered when using `buf` to generate protos. @bufdev hinted that:

> buf is correctly not producing a message value, while protoc is incorrectly producing a default value here

(Note: the issue was found in v1.9.6, and has been adapted to master, which includes the rename to openapiv2)

## To Reproduce

Given the following proto:

```proto
syntax = "proto3";

import "protoc-gen-openapiv2/options/annotations.proto";

package mypackage;

option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
    security: {
        security_requirement: {
            key: "bearer";
        }
    }
};
```

Try to compile the proto with `buf`:

```sh
buf generate --template '{"version":"v1beta1","plugins":[{"name":"openapiv2","out":"out"}]}' --file proto/swagger-description.proto
```

## Expected behavior

The plugin exits successfully.

## Actual Behavior

The plugin panics:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6302f7]

goroutine 1 [running]:
github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger/genswagger.applyTemplate(0xc0001e8080, 0xc0000cc640, 0xc0001c2fe0, 0x0, 0x0)
	/home/hicks/go/pkg/mod/github.com/grpc-ecosystem/grpc-gateway@v1.9.6/protoc-gen-swagger/genswagger/template.go:1177 +0x1017
github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger/genswagger.(*generator).Generate(0xc000010420, 0xc000010610, 0x1, 0x1, 0x1, 0xc000010610, 0x0, 0x1, 0x340)
	/home/hicks/go/pkg/mod/github.com/grpc-ecosystem/grpc-gateway@v1.9.6/protoc-gen-swagger/genswagger/generator.go:116 +0x6d9
main.main()
	/home/hicks/go/pkg/mod/github.com/grpc-ecosystem/grpc-gateway@v1.9.6/protoc-gen-swagger/main.go:117 +0x7eb
plugin swagger: exit status 2
```

The master line that equates to the v1.9.6 panic above is here: https://github.com/grpc-ecosystem/grpc-gateway/blob/master/protoc-gen-openapiv2/internal/genopenapi/template.go#L1358

Note that if the above security_requirement is given `value: {}`, generation with `buf` succeeds.

## Your Environment

gen-go-swagger v1.9.6
GNU/Linux x86_64

--- a/protoc-gen-openapiv2/internal/genopenapi/template.go
+++ b/protoc-gen-openapiv2/internal/genopenapi/template.go
@@ -1355,6 +1355,9 @@ func applyTemplate(p param) (*openapiSwaggerObject, error) {
 			for _, secReq := range spb.Security {
 				newSecReq := openapiSecurityRequirementObject{}
 				for secReqKey, secReqValue := range secReq.SecurityRequirement {
+					if secReqValue == nil {
+						return nil, fmt.Errorf("malformed security requirement spec for key %q; value is required", secReqKey)
+					}
 					newSecReqValue := make([]string, len(secReqValue.Scope))
 					copy(newSecReqValue, secReqValue.Scope)
 					newSecReq[secReqKey] = newSecReqValue

--- a/protoc-gen-openapiv2/internal/genopenapi/template_test.go
+++ b/protoc-gen-openapiv2/internal/genopenapi/template_test.go
@@ -3732,3 +3732,35 @@ func Test_getReservedJsonName(t *testing.T) {
 		})
 	}
 }
+
+func TestParseIncompleteSecurityRequirement(t *testing.T) {
+	swagger := openapi_options.Swagger{
+		Security: []*openapi_options.SecurityRequirement{
+			{
+				SecurityRequirement: map[string]*openapi_options.SecurityRequirement_SecurityRequirementValue{
+					"key": nil,
+				},
+			},
+		},
+	}
+	file := descriptor.File{
+		FileDescriptorProto: &descriptorpb.FileDescriptorProto{
+			SourceCodeInfo: &descriptorpb.SourceCodeInfo{},
+			Name:           proto.String("example.proto"),
+			Package:        proto.String("example"),
+			Options:        &descriptorpb.FileOptions{},
+		},
+	}
+	proto.SetExtension(proto.Message(file.FileDescriptorProto.Options), openapi_options.E_Openapiv2Swagger, &swagger)
+	reg := descriptor.NewRegistry()
+	err := reg.Load(&pluginpb.CodeGeneratorRequest{ProtoFile: []*descriptorpb.FileDescriptorProto{file.FileDescriptorProto}})
+	if err != nil {
+		t.Errorf("failed to reg.Load(): %v", err)
+		return
+	}
+	_, err = applyTemplate(param{File: crossLinkFixture(&file), reg: reg})
+	if err == nil {
+		t.Errorf("applyTemplate(%#v) did not error as expected", file)
+		return
+	}
+}

--- a/protoc-gen-openapiv2/internal/genopenapi/template_test.go
+++ b/protoc-gen-openapiv2/internal/genopenapi/template_test.go
@@ -207,6 +207,9 @@ func TestMessageToQueryParametersWithEnumAsInt(t *testing.T) {
 				Dependency:     []string{},
 				MessageType:    test.MsgDescs,
 				Service:        []*descriptorpb.ServiceDescriptorProto{},
+				Options: &descriptorpb.FileOptions{
+					GoPackage: proto.String(".;example"),
+				},
 			},
 			GoPkg: descriptor.GoPackage{
 				Path: "example.com/path/to/example/example.pb",
@@ -390,6 +393,9 @@ func TestMessageToQueryParameters(t *testing.T) {
 				Dependency:     []string{},
 				MessageType:    test.MsgDescs,
 				Service:        []*descriptorpb.ServiceDescriptorProto{},
+				Options: &descriptorpb.FileOptions{
+					GoPackage: proto.String(".;example"),
+				},
 			},
 			GoPkg: descriptor.GoPackage{
 				Path: "example.com/path/to/example/example.pb",
@@ -512,6 +518,9 @@ func TestMessageToQueryParametersNoRecursive(t *testing.T) {
 				Dependency:     []string{},
 				MessageType:    test.MsgDescs,
 				Service:        []*descriptorpb.ServiceDescriptorProto{},
+				Options: &descriptorpb.FileOptions{
+					GoPackage: proto.String(".;example"),
+				},
 			},
 			GoPkg: descriptor.GoPackage{
 				Path: "example.com/path/to/example/example.pb",
@@ -634,6 +643,9 @@ func TestMessageToQueryParametersRecursive(t *testing.T) {
 				Dependency:     []string{},
 				MessageType:    test.MsgDescs,
 				Service:        []*descriptorpb.ServiceDescriptorProto{},
+				Options: &descriptorpb.FileOptions{
+					GoPackage: proto.String(".;example"),
+				},
 			},
 			GoPkg: descriptor.GoPackage{
 				Path: "example.com/path/to/example/example.pb",
@@ -744,6 +756,9 @@ func TestMessageToQueryParametersWithJsonName(t *testing.T) {
 				Dependency:     []string{},
 				MessageType:    test.MsgDescs,
 				Service:        []*descriptorpb.ServiceDescriptorProto{},
+				Options: &descriptorpb.FileOptions{
+					GoPackage: proto.String(".;example"),
+				},
 			},
 			GoPkg: descriptor.GoPackage{
 				Path: "example.com/path/to/example/example.pb",
@@ -864,6 +879,9 @@ func TestMessageToQueryParametersWellKnownTypes(t *testing.T) {
 					Dependency:     []string{},
 					MessageType:    test.WellKnownMsgDescs,
 					Service:        []*descriptorpb.ServiceDescriptorProto{},
+					Options: &descriptorpb.FileOptions{
+						GoPackage: proto.String("google/well_known"),
+					},
 				},
 				{
 					SourceCodeInfo: &descriptorpb.SourceCodeInfo{},
@@ -872,6 +890,9 @@ func TestMessageToQueryParametersWellKnownTypes(t *testing.T) {
 					Dependency:     []string{"google/well_known.proto"},
 					MessageType:    test.MsgDescs,
 					Service:        []*descriptorpb.ServiceDescriptorProto{},
+					Options: &descriptorpb.FileOptions{
+						GoPackage: proto.String("acme/example"),
+					},
 				},
 			},
 		})
@@ -916,6 +937,9 @@ func TestApplyTemplateSimple(t *testing.T) {
 			Package:        proto.String("example"),
 			MessageType:    []*descriptorpb.DescriptorProto{msgdesc},
 			Service:        []*descriptorpb.ServiceDescriptorProto{svc},
+			Options: &descriptorpb.FileOptions{
+				GoPackage: proto.String(".;example"),
+			},
 		},
 		GoPkg: descriptor.GoPackage{
 			Path: "example.com/path/to/example/example.pb",
@@ -1016,6 +1040,9 @@ func TestApplyTemplateMultiService(t *testing.T) {
 			Package:        proto.String("example"),
 			MessageType:    []*descriptorpb.DescriptorProto{msgdesc},
 			Service:        []*descriptorpb.ServiceDescriptorProto{svc},
+			Options: &descriptorpb.FileOptions{
+				GoPackage: proto.String(".;example"),
+			},
 		},
 		GoPkg: descriptor.GoPackage{
 			Path: "example.com/path/to/example/example.pb",
@@ -1125,6 +1152,9 @@ func TestApplyTemplateOverrideOperationID(t *testing.T) {
 				Package:        proto.String("example"),
 				MessageType:    []*descriptorpb.DescriptorProto{msgdesc},
 				Service:        []*descriptorpb.ServiceDescriptorProto{svc},
+				Options: &descriptorpb.FileOptions{
+					GoPackage: proto.String(".;example"),
+				},
 			},
 			GoPkg: descriptor.GoPackage{
 				Path: "example.com/path/to/example/example.pb",
@@ -1242,7 +1272,9 @@ func TestApplyTemplateExtensions(t *testing.T) {
 				Package:        proto.String("example"),
 				MessageType:    []*descriptorpb.DescriptorProto{msgdesc},
 				Service:        []*descriptorpb.ServiceDescriptorProto{svc},
-				Options:        &descriptorpb.FileOptions{},
+				Options: &descriptorpb.FileOptions{
+					GoPackage: proto.String(".;example"),
+				},
 			},
 			GoPkg: descriptor.GoPackage{
 				Path: "example.com/path/to/example/example.pb",
@@ -1496,6 +1528,9 @@ func TestApplyTemplateRequestWithoutClientStreaming(t *testing.T) {
 			Package:        proto.String("example"),
 			MessageType:    []*descriptorpb.DescriptorProto{msgdesc, nesteddesc},
 			Service:        []*descriptorpb.ServiceDescriptorProto{svc},
+			Options: &descriptorpb.FileOptions{
+				GoPackage: proto.String(".;example"),
+			},
 		},
 		GoPkg: descriptor.GoPackage{
 			Path: "example.com/path/to/example/example.pb",
@@ -1659,6 +1694,9 @@ func TestApplyTemplateRequestWithClientStreaming(t *testing.T) {
 			Package:        proto.String("example"),
 			MessageType:    []*descriptorpb.DescriptorProto{msgdesc, nesteddesc},
 			Service:        []*descriptorpb.ServiceDescriptorProto{svc},
+			Options: &descriptorpb.FileOptions{
+				GoPackage: proto.String(".;example"),
+			},
 		},
 		GoPkg: descriptor.GoPackage{
 			Path: "example.com/path/to/example/example.pb",
@@ -1823,6 +1861,9 @@ func TestApplyTemplateRequestWithUnusedReferences(t *testing.T) {
 			Package:        proto.String("example"),
 			MessageType:    []*descriptorpb.DescriptorProto{reqdesc, respdesc},
 			Service:        []*descriptorpb.ServiceDescriptorProto{svc},
+			Options: &descriptorpb.FileOptions{
+				GoPackage: proto.String(".;example"),
+			},
 		},
 		GoPkg: descriptor.GoPackage{
 			Path: "example.com/path/to/example/example.pb",
@@ -1989,6 +2030,9 @@ func TestApplyTemplateRequestWithBodyQueryParameters(t *testing.T) {
 			Name:           proto.String("book.proto"),
 			MessageType:    []*descriptorpb.DescriptorProto{bookDesc, createDesc},
 			Service:        []*descriptorpb.ServiceDescriptorProto{svc},
+			Options: &descriptorpb.FileOptions{
+				GoPackage: proto.String(".;example"),
+			},
 		},
 		GoPkg: descriptor.GoPackage{
 			Path: "example.com/path/to/book.pb",
@@ -2861,6 +2905,9 @@ func TestSchemaOfField(t *testing.T) {
 				{
 					Name:    proto.String("third_party/google.proto"),
 					Package: proto.String("google.protobuf"),
+					Options: &descriptorpb.FileOptions{
+						GoPackage: proto.String("third_party/google"),
+					},
 					MessageType: []*descriptorpb.DescriptorProto{
 						protodesc.ToDescriptorProto((&structpb.Struct{}).ProtoReflect().Descriptor()),
 						protodesc.ToDescriptorProto((&structpb.Value{}).ProtoReflect().Descriptor()),
@@ -2887,6 +2934,9 @@ func TestSchemaOfField(t *testing.T) {
 					Name:           proto.String("example.proto"),
 					Package:        proto.String("example"),
 					Dependency:     []string{"third_party/google.proto"},
+					Options: &descriptorpb.FileOptions{
+						GoPackage: proto.String(".;example"),
+					},
 					MessageType: []*descriptorpb.DescriptorProto{
 						{
 							Name: proto.String("Message"),
@@ -3183,6 +3233,9 @@ func TestRenderMessagesAsDefinition(t *testing.T) {
 					MessageType:    test.msgDescs,
 					EnumType:       []*descriptorpb.EnumDescriptorProto{},
 					Service:        []*descriptorpb.ServiceDescriptorProto{},
+					Options: &descriptorpb.FileOptions{
+						GoPackage: proto.String(".;example"),
+					},
 				},
 				Messages: msgs,
 			}
@@ -3517,6 +3570,9 @@ func TestMessageOptionsWithGoTemplate(t *testing.T) {
 					MessageType:    test.msgDescs,
 					EnumType:       []*descriptorpb.EnumDescriptorProto{},
 					Service:        []*descriptorpb.ServiceDescriptorProto{},
+					Options: &descriptorpb.FileOptions{
+						GoPackage: proto.String(".;example"),
+					},
 				},
 				Messages: msgs,
 			}
@@ -3584,6 +3640,9 @@ func TestTemplateWithoutErrorDefinition(t *testing.T) {
 			Package:        proto.String("example"),
 			MessageType:    []*descriptorpb.DescriptorProto{msgdesc},
 			Service:        []*descriptorpb.ServiceDescriptorProto{svc},
+			Options: &descriptorpb.FileOptions{
+				GoPackage: proto.String(".;example"),
+			},
 		},
 		GoPkg: descriptor.GoPackage{
 			Path: "example.com/path/to/example/example.pb",
@@ -3748,7 +3807,9 @@ func TestParseIncompleteSecurityRequirement(t *testing.T) {
 			SourceCodeInfo: &descriptorpb.SourceCodeInfo{},
 			Name:           proto.String("example.proto"),
 			Package:        proto.String("example"),
-			Options:        &descriptorpb.FileOptions{},
+			Options: &descriptorpb.FileOptions{
+				GoPackage: proto.String(".;example"),
+			},
 		},
 	}
 	proto.SetExtension(proto.Message(file.FileDescriptorProto.Options), openapi_options.E_Openapiv2Swagger, &swagger)

